### PR TITLE
Replace gostats with tally

### PR DESCRIPTION
**Pull request by nishkrishnan**
Basically I create a `Metrics` object in `valid.GlobalCfg` which contains statsd info. This object should technically be parsed from yaml config and should eventually be used to support prometheus or any other metrics provider. (This will be done in a followup PR to not blow up the scope)

I also added runtime stats publishing to our executor service. This means that when we OSS this change, we'll need to move executor service out from the lyft package.

This is the first PR to get this out of the door and ensure our stats are working as expected.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -149,8 +152,8 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		return nil, err
 	}
 
-	statsScope := stats.NewDefaultStore().Scope(userConfig.StatsNamespace)
-	statsScope.Store().AddStatGenerator(stats.NewRuntimeStats(statsScope.Scope("go")))
+	// statsScope := stats.NewDefaultStore().Scope(userConfig.StatsNamespace)
+	// statsScope.Store().AddStatGenerator(stats.NewRuntimeStats(statsScope.Scope("go")))
```
Comment (msarvar) [inline server/server.go:156]: delete?

File: server/server.go [hunk 2]
```diff
@@ -917,7 +932,10 @@ func (s *Server) Start() error {
 	s.waitForDrain()
 
 	// flush stats before shutdown
-	s.StatsScope.Store().Flush()
+	err := s.StatsCloser.Close()
+	if err != nil {
```
Comment (msarvar) [inline server/server.go:936]: nit:
```suggestion
	if err := s.StatsCloser.Close(); err != nil {
```

File: server/metrics/statsd.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/uber-go/tally"
+	tallystatsd "github.com/uber-go/tally/statsd"
+)
+
+func NewLoggingScope(logger logging.SimpleLogging, statsNamespace string) (tally.Scope, io.Closer, error) {
```
Comment (msarvar) [inline server/metrics/statsd.go:16]: This seems to only be used by tests? Can you leave a comment about it?

File: server/metrics/statsd.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	scope, closer := tally.NewRootScope(tally.ScopeOptions{
+		Prefix:   statsNamespace,
+		Reporter: reporter,
+	}, time.Second)
+
+	return scope, closer, nil
+}
+
+func newReporter(cfg valid.Metrics, logger logging.SimpleLogging) (tally.StatsReporter, error) {
+	if cfg.Statsd == nil {
```
Comment (msarvar) [inline server/metrics/statsd.go:47]: Is this always the case? What if there was a misconfigation that somehow set the statsd to nil? Shouldn't we return an error here? Looks like `newLoggingReporter` is only used by tests.

**Review by msarvar — COMMENTED** [review]
Looks good, couple comments.

File: server/metrics/statsd.go (hunk 3 above)
Comment (nishkrishnan) [reply server/metrics/statsd.go:16]: It'll be used by the local server setup as well if it's running in debug mode.

File: server/metrics/statsd.go (hunk 4 above)
Comment (nishkrishnan) [reply server/metrics/statsd.go:47]: the idea is here is to have a set of fields in that config object for statsd, prometheus and really any time of metrics supported by tally.

We are checking to see which metrics object is populated and if nothing is, we just return the logging scope.

**Review by msarvar — APPROVED** [review]

**Pull request merged**